src/main/java/org/alliancegenome/curation_api/jobs/executors/Gff3Executor.java

```diff
@@ -0,0 +1,190 @@
+package org.alliancegenome.curation_api.jobs.executors;
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+
+import org.alliancegenome.curation_api.enums.BackendBulkDataProvider;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
+import org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData;
+import org.alliancegenome.curation_api.jobs.util.CsvSchemaBuilder;
+import org.alliancegenome.curation_api.model.entities.GenomeAssembly;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkFMSLoad;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFile;
+import org.alliancegenome.curation_api.model.entities.bulkloads.BulkLoadFileHistory;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.Gff3DTO;
+import org.alliancegenome.curation_api.response.APIResponse;
+import org.alliancegenome.curation_api.response.LoadHistoryResponce;
+import org.alliancegenome.curation_api.services.CodingSequenceService;
+import org.alliancegenome.curation_api.services.ExonService;
+import org.alliancegenome.curation_api.services.Gff3Service;
+import org.alliancegenome.curation_api.services.TranscriptService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
+import org.apache.commons.lang3.ObjectUtils;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class Gff3Executor extends LoadFileExecutor {
+
+	@Inject Gff3Service gff3Service;
+	@Inject ExonService exonService;
+	@Inject CodingSequenceService cdsService;
+	@Inject TranscriptService transcriptService;
+	
+	public void execLoad(BulkLoadFile bulkLoadFile) {
+		try {
+
+			CsvSchema gff3Schema = CsvSchemaBuilder.gff3Schema();
+			CsvMapper csvMapper = new CsvMapper();
+			MappingIterator<Gff3DTO> it = csvMapper.enable(CsvParser.Feature.INSERT_NULLS_FOR_MISSING_COLUMNS).readerFor(Gff3DTO.class).with(gff3Schema).readValues(new GZIPInputStream(new FileInputStream(bulkLoadFile.getLocalFilePath())));
+			List<Gff3DTO> gffData = it.readAll();
```

Comment:
@markquintontulloch this is going to be a problem for large GFF files.

Reply:
They aren't that big compared to a lot of files we process